Mathlib/AlgebraicGeometry/Scheme.lean

```diff
@@ -58,19 +58,31 @@ abbrev Opens (X : Scheme) : Type* := TopologicalSpace.Opens X
 
 /-- A morphism between schemes is a morphism between the underlying locally ringed spaces. -/
 -- @[nolint has_nonempty_instance] -- Porting note(#5171): linter not ported yet
-def Hom (X Y : Scheme) : Type* :=
-  X.toLocallyRingedSpace ⟶ Y.toLocallyRingedSpace
+structure Hom (X Y : Scheme) extends X.toLocallyRingedSpace.Hom Y.toLocallyRingedSpace where
+
+/-- See Note [custom simps projection] -/
+abbrev Hom.val {X Y : Scheme.{u}} (f : X.Hom Y) :
+    X.toLocallyRingedSpace ⟶ Y.toLocallyRingedSpace :=
+  f.toHom_1
```

Comment:
I think this is quite stable. On a side note, do you think `Hom.val` should get a more descriptive name like `toLRSHom`?

Reply:
I think `toLRSHom` is too long. Is `Hom.lrs` too cryptic? If not we could have `Hom.psh` for the underlying `Hom` of presheafed spaces.

Reply:
I think the current design of the library shouldn't require user to use these type-cast that often so it's fine if it's a bit long?

Reply:
Sounds reasonable, since `f.c` and `f.base` are directly accessible now, I have not checked in detail though how often the explicit cast is needed. Maybe we can replace `val` as you suggested by `toLRSHom` and add `toPSSHom` as `abbrev` for `toHom`. If it turns out to be annoying we can `sed` back quickly.

Reply:
I am happy with the `val -> toLRSHom` and `toHom -> toShHom`.